cobbler/actions/buildiso/__init__.py

```diff
@@ -0,0 +1,246 @@
+"""
+Builds bootable CD images that have PXE-equivalent behavior for all Cobbler distros/profiles/systems currently in
+memory.
+"""
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+# SPDX-FileCopyrightText: Copyright 2006-2009, Red Hat, Inc and Others
+# SPDX-FileCopyrightText: Michael DeHaan <michael.dehaan AT gmail>
+
+import logging
+import os
+import os.path
+import pathlib
+import shutil
+from typing import Dict, List, Optional, Union
+
+from cobbler import utils
+
+
+def add_remaining_kopts(kopts: dict) -> str:
+    """Add remaining kernel_options to append_line
+
+    :param kopts: The kernel options which are not present in append_line.
+    :return: A single line with all kernel options from the dictionary in the string. Starts with a space.
+    """
+    append_line = [""]  # empty str to ensure the returned str starts with a space
+    for option, args in kopts.items():
+        if args is None:
+            append_line.append(f"{option}")
+            continue
+
+        if not isinstance(args, list):
+            args = [args]
+
+        for arg in args:
+            arg_str = format(arg)
+            if ' ' in arg_str:
+                arg_str = f'"{arg_str}"'
```

Comment:
BTW, changing the single quotes to double quotes was on purpose, based on 

> Double-quotes can be used to protect spaces in values, e.g.:

from https://www.kernel.org/doc/html/v4.14/admin-guide/kernel-parameters.html#the-kernel-s-command-line-parameters